### PR TITLE
Added tests to durable executor service for full ring-buffer case and newly joined member case

### PR DESCRIPTION
**Pull request by gurbuzali**

### CODE REVIEW

File: hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableExecutorServiceTest.java [hunk 1]
```diff
@@ -188,6 +188,30 @@ public void onFailure(Throwable t) {
...
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                if (t.getCause() instanceof RejectedExecutionException) {
+                    latch.countDown();
+                }
+            }
+        });
+        assertOpenEventually(latch, 10);
```
Comment (mdogan) [inline hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableExecutorServiceTest.java]: `assertOpenEventually(latch, 10)` can fail under high load, in slow environment. I think better is to just leave default timeout...
`assertOpenEventually(latch)`

**Comment by mdogan** [comment]
👍

File: hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java [hunk 2]
```diff
@@ -26,9 +26,6 @@
 
     private transient Object result;
 
-    public RetrieveAndDisposeResultOperation() {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java]: is this operation always running locally? (=it is never (de-)serialized?)

File: hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableExecutorServiceTest.java (hunk 1 above)
Comment (gurbuzali) [inline hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableExecutorServiceTest.java]: ok, I will fix it after merge

File: hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java (hunk 2 above)
Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java]: no it can be a remote operation

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java]: so shouldn't it have an argument-less contructor?

Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/operations/RetrieveAndDisposeResultOperation.java]: it extends `DisposeResultOperation` which has an argument-less contructor

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**